frontend/src/metabase/common/components/FilterPicker/NumberFilterPicker.tsx

```diff
@@ -1,31 +1,215 @@
+
+  const handleOperatorChange = (newOperatorName: Lib.FilterOperatorName) => {
+    setOperatorName(newOperatorName);
+    setValues([]);
+  };
+
+  const handleValuesChange = (values: number[]) => {
+    setValues(values);
+  };
+
+  const handleFilterChange = () => {
+    if (operatorName && values.length) {
+      const operator = Lib.findFilterOperator(
+        query,
+        stageIndex,
+        column,
+        operatorName,
+      );
+      if (operator) {
+        onChange(
+          Lib.numberFilterClause(query, stageIndex, {
+            operator,
+            column,
+            values,
+          }),
+        );
+        onClose();
+      }
+    }
+  };
+
+  const numberValueType = getNumberValueType(operatorName);
+  const isFilterValid = isNumberFilterValid(operatorName, values);
+
   return (
     <>
       <Header>
         <BackButton onClick={onBack}>{columnName}</BackButton>
+        <FilterOperatorPicker
+          query={query}
+          stageIndex={stageIndex}
+          column={column}
+          value={operatorName}
+          onChange={handleOperatorChange}
+        />
       </Header>
-      <Flex p="sm">
-        <Text>Number editor UI</Text>
-      </Flex>
-      <Footer>
+      <NumberValueInput
+        values={values}
+        onChange={handleValuesChange}
+        type={numberValueType}
+        column={column}
+      />
+      <Footer
+        mt={
+          numberValueType === "none" ? -1 : undefined
+        } /* to collapse borders */
+      >
         <Box />
-        <Button>{filter ? t`Update filter` : t`Add filter`}</Button>
+        <Button disabled={!isFilterValid} onClick={handleFilterChange}>
+          {filter ? t`Update filter` : t`Add filter`}
+        </Button>
       </Footer>
     </>
   );
 }
+
+function NumberValueInput({
+  values,
+  onChange,
+  type,
+  column,
+}: {
+  values: number[];
+  onChange: (values: number[]) => void;
+  type: NumberValueType;
+  column: Lib.ColumnMetadata;
+}) {
+  const placeHolder = t`Enter a number`;
+  const fieldId = useMemo(() => Lib._fieldId(column), [column]);
+
+  // const prefix = '$'; TODO
+
+  switch (type) {
+    case "multiple":
+      return (
+        <Flex p="md" mah={300} style={{ overflowY: "scroll" }}>
+          <FieldValuesWidget
+            fields={[new Field({ id: fieldId })]}
+            className="input"
+            value={values}
+            minWidth={"300px"}
+            onChange={onChange}
+            placeholder={placeHolder}
+            disablePKRemappingForSearch
+            autoFocus
+            multi={type === "multiple"}
+          />
+        </Flex>
+      );
+    case "single":
+      return (
+        <Flex p="md">
+          <NumberInput
+            value={values[0]}
+            onChange={(newValue: number) => onChange([newValue])}
+            placeholder={placeHolder}
+            autoFocus
+          />
+        </Flex>
+      );
```

Comment:
In the DefaultPicker, we use `FieldValuesWidget` with search disabled for single value number filters. that seemed goofy to me, might as well use the simpler component if we're not using anything from the FieldValuesWidget.